R/utils.R

```diff
@@ -25,6 +25,14 @@ if (base::getRversion() < "3.2.0") {  # Apr 2015
   isNamespaceLoaded = function(x) x %chin% loadedNamespaces()
 }
 
+# R 3.3.0 [April 2016]
+if (!exists('startsWith', as.environment('package:base'))) {
+  startsWith = function(x, stub) substr(x, 1L, nchar(stub)) == stub
+}
+if (!exists('endsWith', as.environment('package:base'))) {
+  endsWith = function(x, stub) {n = nchar(x); substr(x, n-nchar(stub)+1L, n) == stub}
+}
```

Comment:
I would add extra tests for vector input, len 0 input, NA, and some non native encoding